pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -102,37 +119,62 @@ func LowNodeUtilization(ctx context.Context, client clientset.Interface, strateg
 	klog.V(1).Infof("Total number of pods evicted: %v", podEvictor.TotalEvicted())
 }
 
-func validateThresholds(thresholds api.ResourceThresholds) bool {
+// validateStrategyConfig checks if the strategy's config is valid
+func validateStrategyConfig(thresholds, targetThresholds api.ResourceThresholds) error {
+	// validate thresholds and targetThresholds config
+	if err := validateThresholds(thresholds); err != nil {
+		return fmt.Errorf("thresholds config is not valid: %v", err)
+	}
+	if err := validateThresholds(targetThresholds); err != nil {
+		return fmt.Errorf("targetThresholds config is not valid: %v", err)
+	}
+
+	// validate if thresholds and targetThresholds have same resources configured
+	if len(thresholds) != len(targetThresholds) {
+		return fmt.Errorf("thresholds and targetThresholds configured different resources")
+	}
+	for resourceName, value := range thresholds {
+		if targetValue, ok := targetThresholds[resourceName]; !ok {
+			return fmt.Errorf("thresholds and targetThresholds configured different resources")
+		} else if value > targetValue {
+			return fmt.Errorf("thresholds' %v percentage is greater than targetThresholds'", resourceName)
+		}
+	}
+	return nil
+}
+
+// validateThresholds checks if thresholds have valid resource name and resource percentage configured
+func validateThresholds(thresholds api.ResourceThresholds) error {
 	if thresholds == nil || len(thresholds) == 0 {
-		klog.V(1).Infof("no resource threshold is configured")
-		return false
+		return fmt.Errorf("no resource threshold is configured")
 	}
 	for name := range thresholds {
 		switch name {
```

Comment:
Given all valid resources execute the same code, you can consolidate them:
```
switch name {
case v1.ResourceCPU:
case v1.ResourceMemory:
case v1.ResourcePods:
	if !validateResourcePercentage(thresholds[name]) {
		return fmt.Errorf("%v threshold not valid", name)
	}
default:
	return fmt.Errorf("only cpu, memory, or pods thresholds can be specified")
}
```
Also, given `validateResourcePercentage` gets to used only once, you can inline its implementation:

```
switch name {
case v1.ResourceCPU:
case v1.ResourceMemory:
case v1.ResourcePods:
	percent := thresholds[name]
	if percent < MinResourcePercentage || percent > MaxResourcePercentage {
		return fmt.Errorf("%v threshold not in [%v, %v] range", name, MinResourcePercentage, MaxResourcePercentage)
	}
default:
	return fmt.Errorf("only cpu, memory, or pods thresholds can be specified")
}
```

Reply:
Wow,that's much better!